digitiser-aggregator/src/spanned_frame.rs

```diff
@@ -0,0 +1,64 @@
+use std::{fmt::Debug, time::Duration};
+use supermusr_common::tracer::Spanned;
+use supermusr_streaming_types::FrameMetadata;
+use tracing::{trace_span, Span};
+
+use crate::{
+    data::Accumulate,
+    frame::{AggregatedFrame, AggregatedFrameLike, FrameCache, PartialFrame, PartialFrameLike},
+};
+
+pub(crate) type SpannedPartialFrame<D> = Spanned<PartialFrame<D>>;
+
+impl<D: Debug> PartialFrameLike<D> for SpannedPartialFrame<D> {
+    fn new(ttl: Duration, metadata: FrameMetadata) -> Self {
+        let span = trace_span!("Frame");
+        Spanned {
+            span,
+            value: PartialFrame::<D>::new(ttl, metadata),
+        }
+    }
+}
+
+pub(crate) struct SpannedAggregatedFrame<D> {
```

Comment:
Why can't `AggregatedFrame` also use `Spanned`?

Reply:
Due to Rust rules on implementing foreign traits on foreign types. I'm going to redesign this in a subsequent PR however.

Reply:
I don't see any reason not to have the trait impl next to the type definition in this case though. The defining crate is in this repo and the fact there is an additional trait on the type should not adversely affect any other crate.

Reply:
It's a restriction on foreign types *and* foreign traits, so as Spanned<T> is from another crate, it won't allow compilation.
That is
```rust
SpannedAggregatedFrame<D> = Spanned<AggregatedFrame<D>>;

impl<D: Debug> AggregatedFrameLike<D> for Spanned<AggregatedFrame<D>> {
...
}
```
is not allowed.

Reply:
Ah, I see, it's the `impl From` that is the issue. If this is going to be redesigned then it is fine for now.